### DISCUSSION TITLE
Return Kubernetes Events for a certain task

### DISCUSSION THREAD

**Discussion by robert-ulbrich-mercedes-benz** [flyteorg/flyte#5089] [RFC Incubator]
**Motivation**

Often tasks are displayed as running, like in the screenshot shown below: 

![image](https://github.com/flyteorg/flyte/assets/141313113/17548ae0-8a14-452c-8709-fe317af7a9a9)

In reality the pod cannot be started, because the image is invalid and cannot be pulled. The user should get some feedback in the field that is highlighted. It looks like it is coming from task executions API endpoint. The data seems to be coming from a database table and is outdated. 

Users are therefore often confused, why their workloads do not complete despite being shown running for a long time.

**Functionality**

The field providing feedback about the status of the task should also contain relevant [Kubernetes Events](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/) that are kept up to date to provide meaningful feedback.

**Comment by fg91** [comment]
I think the reason why the event is often outdated for several plugins is that the phase version is only increased consistently in the pod plugin (see [here](https://github.com/flyteorg/flyte/blob/c81133bfc79e8eb2efb729a5f6e3f972195967e4/flyteplugins/go/tasks/plugins/k8s/pod/plugin.go#L245)). If you don't increase the version when the "reason changes", no new event is sent to propeller.

**Reply by kumare3** [reply]
Yes lets update the phasing system. Sending k8s events back is dangerous. This can be a lot of data and is not optimal. Most users will not have a very complicated setup and to serve most users, we need to keep it simple, scalable and easy

**Comment by davidmirror-ops** [comment]
2024-04-25 Contributors meetup notes: do not proceed to RFC or implementation as there are UX concerns.
